### PR TITLE
Coins enhancement

### LINKED ISSUE #446: Enhance Coins Component
# **Description**
**Task:** Add text and sound effects for when coins are collected
**Feature:** #410 

Add sound effects for when player collects coins, for futher enhancement. Moreover,  to make the game more interactive, inform player how much coins they collect by defeating each animal.
 
## **Example (optional)**
When player successfully defeats an animal, a text is displayed on the screen to inform player how many coins they earned along with some sound effects

# **Dependencies**
- [x] #296 (#xx)

# **Milestones**
List of steps that need to be completed for this task.
- [x] enhance the current implementation (Oct 12th)
- [x] add the text to UI (Oct 12th)
- [x] Add sound effects (October 12th)
- [x] Add tests if needed (October 14th)
- [x] Documentation (Oct 14th)
 
**Completion Deadline:** Oct 16th

 
# **Documentation**
- [Scoring System](https://github.com/UQcsse3200/2024-studio-1/wiki/Scoring-System)
 
# **Member**
- Manya Khosla @manya-k

### LINKED ISSUE #448: Loading health bar
## Description
When the game is saved, the player's health bar set the max health of player to be the previously saved health not the max health of player

## How To Reproduce
Steps to reproduce the behaviour:
1. start the game
2. take damage
3. save the game
4. load the game
5. see the difference between the damage amount and player's health bar

## Expected Behaviour
when the game is loaded, the red colour of health bar must health epresent the capacity of player

## Screenshots
image

## Temporary Fix
ignore since it is not a game breaking bug

### PR DESCRIPTION
**Pull request by manya-k**
# Description

Refactored a bit of CoinsComponent so that it does not take inventory anymore as it was not necessary 
Added a label to show how many coins each animals give
Added sound effects for when player collects the coin

Also fixed player health bar bug: previously after taking damaged if the game was saved, upon loading, max health of player displayed on the healthbar was set to saved current health. 

Closes #446 
Fixes #448 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Manual Testing
- [x] Junit Testing

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Aryan-S-Gupta — APPROVED** [review]

**Comment by manya-k** [comment]
> The bug fix in the CombatStatsComponent looks good. Ensure that the changes are tested thoroughly to confirm they resolve the issue without introducing new ones. Nice work!

thanks! they have been tested

File: source/core/src/main/com/csse3200/game/components/player/PlayerCoinDisplay.java [hunk 1]
```diff
@@ -63,6 +69,31 @@ private void updateCoinLabel() {
...
+     */
+    private void showCoins(int coins) {
+        ServiceLocator.getResourceService().playSound("sounds/collectCoin.mp3");
+        String text = String.format("+%d coins added", coins);
+        Label coinText = new Label(text, skin, "small");
+        Table table = new Table();
+        table.top().padTop(150f);
+        table.setFillParent(true);
+        table.row().padTop(5f);
+        table.add(coinText).padTop(5f);
```
Comment (IanP96) [inline source/core/src/main/com/csse3200/game/components/player/PlayerCoinDisplay.java:86]: you can probably just sum these padding vals up and just add it in one place, this should work too

File: source/core/src/main/com/csse3200/game/components/player/PlayerCoinDisplay.java [hunk 2]
```diff
@@ -63,6 +69,31 @@ private void updateCoinLabel() {
...
+        stage.addActor(table);
+        // unrender the label after 3 second of display
+        Timer.schedule(new Timer.Task() {
+            @Override
+            public void run() {
+                coinText.remove();
+            }
+        }, 1f);
```
Comment (IanP96) [inline source/core/src/main/com/csse3200/game/components/player/PlayerCoinDisplay.java:94]: 3 seconds or 1 second?

**Review by IanP96 — APPROVED** [review]
looks good, just make sure you've done enough manual testing as I imagine this would be hard to test automatically

File: source/core/src/main/com/csse3200/game/components/player/PlayerCoinDisplay.java (hunk 2 above)
Comment (manya-k) [reply source/core/src/main/com/csse3200/game/components/player/PlayerCoinDisplay.java:94]: 3 seconds ends up being too long

Comment (manya-k) [reply source/core/src/main/com/csse3200/game/components/player/PlayerCoinDisplay.java:94]: check it out with 3 seconds and if you prefer that, i can change it

**Review by vsinh14 — APPROVED** [review]
Tested and all code changes look good

**Review by dhruvanil55 — APPROVED** [review]
LGTM

**Review by dhruvanil55 — APPROVED** [review]
LGTM

**Pull request merged**